src/waveform/renderers/allshader/waveformrendererslipmode.cpp

```diff
@@ -0,0 +1,107 @@
+#include "waveform/renderers/allshader/waveformrendererslipmode.h"
+
+#include <QDomNode>
+#include <memory>
+
+#include "control/controlproxy.h"
+#include "waveform/renderers/waveformwidgetrenderer.h"
+#include "waveform/waveformwidgetfactory.h"
+#include "widget/wskincolor.h"
+
+namespace {
+
+constexpr int kBlinkingPeriodMillis = 1600;
+constexpr float positionArray[] = {-1.f, -1.f, 1.f, -1.f, -1.f, 1.f, 1.f, 1.f};
+
+} // anonymous namespace
+
+namespace allshader {
+
+WaveformRendererSlipMode::WaveformRendererSlipMode(
+        WaveformWidgetRenderer* waveformWidget)
+        : WaveformRenderer(waveformWidget),
+          m_pSlipMode(nullptr),
+          m_slipBorderTopOutlineSize(10.f),
+          m_slipBorderBottomOutlineSize(10.f) {
+}
+
+bool WaveformRendererSlipMode::init() {
+    m_timer.restart();
+
+    m_pSlipMode.reset(new ControlProxy(
+            m_waveformRenderer->getGroup(), "slip_enabled"));
+
+    return true;
+}
+
+void WaveformRendererSlipMode::setup(const QDomNode& node, const SkinContext& context) {
+    m_color = QColor(224, 224, 224);
+    const QString slipModeOutlineColorName = context.selectString(node, "SlipBorderOutlineColor");
+    if (!slipModeOutlineColorName.isNull()) {
+        m_color = QColor(slipModeOutlineColorName);
+        m_color = WSkinColor::getCorrectColor(m_color);
```

Comment:
Writing the member twice is confusing here. How about: 
```suggestion
        m_color = WSkinColor::getCorrectColor(QColor(slipModeOutlineColorName));
```
Or introduce another temporary.